src/OpenSage.Game/Client/Drawable.cs

```diff
@@ -363,6 +395,1110 @@ internal void BuildRenderList(RenderList renderList, Camera camera, in TimeInter
+        var uniformAxialDamping = locomotor.LocomotorTemplate.UniformAxialDamping;
+        var maxSuspensionExtension = locomotor.LocomotorTemplate.MaximumWheelExtension;
+        var wheelAngle = locomotor.LocomotorTemplate.FrontWheelTurnAngle;
+        var hasSuspension = locomotor.LocomotorTemplate.HasSuspension;
+
+        // get object from logic
+        var obj = GameObject;
+
+        var ai = obj.AIUpdate;
+        if (ai == null)
+        {
+            return;
+        }
+
+        // get object physics state
+        var physics = obj.Physics;
+        if (physics == null)
+        {
+            return;
+        }
+
+        // get our position and direction vector
+        var pos = Translation;
+        var dir = UnitDirectionVector2D;
+        var accel = physics.Acceleration;
+
+        // compute perpendicular (2d)
+        var perp = new Vector3(
+            -dir.Y,
+            dir.X,
+            0.0f);
+
+        // find pitch and roll of terrain under chassis
+        var hheight = _gameEngine.Game.TerrainLogic.GetLayerHeight(pos.X, pos.Y, obj.Layer, out var normal);
+
+        var dot = normal.X * dir.X + normal.Y * dir.Y;
+        var groundPitch = dot * (MathF.PI / 2.0f);
+
+        dot = normal.X * perp.X + normal.Y * perp.Y;
+        var groundRoll = dot * (MathF.PI / 2.0f);
+
+        var airborne = obj.IsSignificantlyAboveTerrain;
+
+        if (airborne)
+        {
+            if (hasSuspension)
+            {
+                // Wheels extend when airborne.
+                _locomotorInfo.WheelInfo.FramesAirborne = 0;
+                _locomotorInfo.WheelInfo.FramesAirborneCounter++;
+                if (pos.Z - hheight > -maxSuspensionExtension)
+                {
+                    _locomotorInfo.WheelInfo.RearLeftHeightOffset += (maxSuspensionExtension - _locomotorInfo.WheelInfo.RearLeftHeightOffset) / 2.0f;
+                    _locomotorInfo.WheelInfo.RearRightHeightOffset = _locomotorInfo.WheelInfo.RearLeftHeightOffset;
+                }
+                else
+                {
+                    _locomotorInfo.WheelInfo.RearLeftHeightOffset += (0 - _locomotorInfo.WheelInfo.RearLeftHeightOffset) / 2.0f;
+                    _locomotorInfo.WheelInfo.RearRightHeightOffset = _locomotorInfo.WheelInfo.RearLeftHeightOffset;
+                }
+            }
+            // Calculate suspension info.
+            var length2 = obj.Geometry.MajorRadius;
+            //Real width = obj->getGeometryInfo().getMinorRadius();
+            var pitchHeight2 = length2 * MathF.Sin(_locomotorInfo.Pitch + _locomotorInfo.AccelerationPitch - groundPitch);
+            //Real rollHeight = width*Sin(_locomotorInfo.Roll + _locomotorInfo.AccelerationRoll - groundRoll);
+            info.TotalZ = MathF.Abs(pitchHeight2) / 4;// + MathF.Abs(rollHeight)/4;
+            //return; // maintain the same orientation while we fly through the air.
```

Comment:
do we not want to return here?

Reply:
Correct, for motorcycles, we don't return here. Which means that motorcycles, unlike four-wheeled vehicles, change their pitch and roll while in the air. Presumably that's intentional.